GitHub workflows & toctree

MKhalusova opened the pull request:
The PR adds 
* the GitHub workflows needed to build the cookbook under hf.co/learn and to preview PRs.
* _toctree.yml

MKhalusova commented:
@mishig25 Should I also move the notebook files under `notebooks/en` instead of them being under `notebooks`?

mishig25 commented:
>@mishig25 Should I also move the notebook files under notebooks/en instead of them being under notebooks?

yes if we expect there is gonna be translations

MKhalusova commented:
Moved the notebooks

MKhalusova commented:
@mishig25 I added a `_toctree.yaml`. Two questions
 - Should I leave the `.ipynb` or remove it? For `.mdx` files we don't specify it. 
 - I want to add an index page. Is it ok to have it as a `.mdx` file?

mishig25 commented:
> Should I leave the .ipynb or remove it? For .mdx files we don't specify it.

remove it

> I want to add an index page. Is it ok to have it as a .mdx file?

Yes, you can actually have it just in .md rather than .mdx. It doesnt make difference

mishig25 commented on notebooks/en/_toctree.yml:
```diff
@@ -0,0 +1,12 @@
+- title: Open-Source AI Cookbook
+  sections:
+  - local: FAISS_with_HF_datasets_and_CLIP
+    title: Embedding multimodal data for similarity search using 🤗 transformers, 🤗 datasets and FAISS
```
Small nit: but I think titles should be usually shorter. Otherwise, they would look too long on the navigation side bar

MKhalusova commented:
@mishig25 I've updated the titles in the TOC to make them shorter, and added an index page. 

I can add the content for the index page in the separate PR so that we could merge this one sooner and work on things in parallel.

mishig25 commented:
Great! I can continue on this PR bit

MKhalusova commented:
Updated this PR with `main` (notebook outputs)

mishig25 commented:
using https://github.com/huggingface/doc-builder/pull/469
